core/internal/logger.hpp

```diff
@@ -0,0 +1,61 @@
+/**
+ * This is the header for the logger.
+ * Very little of the logger will be exposed through the API.
+ * Author: Jake Roggenbuck
+ * 10-08-23
+ */
+
+#pragma once
+
+#include <iostream>
+
+namespace jml {
+
+enum Severity { DEBUG, INFO, WARN, ERR, FATAL };
+
+static const std::string severity_text[] = {"DEBUG", "INFO", "WARN", "ERR",
+                                            "FATAL"};
+
+const std::string get_severity_text(enum Severity s);
+
+class Log {
+  private:
+    enum Severity severity;
+    std::string message;
+
+  public:
+    Log &operator<<(enum Severity s);
+    Log &operator<<(std::string m);
+    Log &operator<<(int m);
+
+    enum Severity get_severity();
+    std::string get_message();
+
+    Log();
```

Comment:
I just remembered that you can do default parameters just like in Python rather than overloading, so we should probably do that instead.